ocaml/xapi/xapi_cluster_host.ml

```diff
@@ -79,6 +79,8 @@ let join_internal ~__context ~self =
 
       debug "Enabling clusterd and joining cluster_host %s" (Ref.string_of self);
       Xapi_clustering.Daemon.enable ~__context;
+      Db.Cluster_host.set_joined ~__context ~self ~value:true;
```

Comment:
Please move these after the call to join: the call to join might fail and we want to set joined to true only if we actually succeeded.
So match on result Ok/Error should be inhere as well

Reply:
So `resync_host` just calls `join_internal`, then continues the `if enabled then` block? That works better actually.